File: predicateMatrix/src/main/java/org/sqlunet/predicatematrix/provider/PredicateMatrixControl.java
package org.sqlunet.predicatematrix.provider;

/**
 * PredicateMatrix query control
 *
 * @author <a href="mailto:1313ou@gmail.com">Bernard Bou</a>
 */
public class PredicateMatrixControl
{
	// table codes
	static protected final int PM = 10;

	// join codes
	static protected final int PM_X = 11;

	public static Result queryMain(final int code, final String uriLast, final String[] projection0, final String selection0, final String[] selectionArgs0)
	{
		String table;
		String groupBy = null;

		switch (code)
		{
			// I T E M
			// the incoming URI was for a single item because this URI was for a single row, the _ID value part is present.
			// get the last path segment from the URI: this is the _ID value. then, append the value to the WHERE clause for the query

			// J O I N S

			case PredicateMatrixControl.PM:
				table = Q.PM.TABLE;
				break;

			case PredicateMatrixControl.PM_X:
				table = Q.PM_X.TABLE;
				break;

			default:
				return null;
		}
		return new Result(table, projection0, selection0, selectionArgs0, groupBy);
	}

	static public class Result
	{
		final String table;
		final String[] projection;
		final String selection;
		final String[] selectionArgs;
		final String groupBy;

		public Result(final String table, final String[] projection, final String selection, final String[] selectionArgs, final String groupBy)
		{
			this.table = table;
			this.projection = projection;
			this.selection = selection;
			this.selectionArgs = selectionArgs;
			this.groupBy = groupBy;
		}
	}
}

File: predicateMatrix/src/main/java/org/sqlunet/predicatematrix/provider/PredicateMatrixControl.kt
/*
 * Copyright (c) 2023. Bernard Bou
 */
package org.sqlunet.predicatematrix.provider

/**
 * PredicateMatrix query control
 *
 * @author [Bernard Bou](mailto:1313ou@gmail.com)
 */
object PredicateMatrixControl {

    // table codes
    const val PM = 10

    // join codes
    const val PM_X = 11

    fun queryMain(code: Int, projection0: Array<String>?, selection0: String?, selectionArgs0: Array<String>?): Result? {
        val groupBy: String? = null
        val table: String = when (code) {
            PM -> Q.PM.TABLE
            PM_X -> Q.PM_X.TABLE
            else -> return null
        }
        return Result(table, projection0, selection0, selectionArgs0, groupBy)
    }

    data class Result(val table: String, val projection: Array<String>?, val selection: String?, val selectionArgs: Array<String>?, val groupBy: String?) {

        override fun toString(): String {
            return "table='$table'\nprojection=${projection.contentToString()}\nselection='$selection'\nselectionArgs=${selectionArgs.contentToString()}\ngroupBy=$groupBy"
        }
    }
}
